unitycatalog-ai/integrations/crewai/src/ucai_crewai/toolkit.py

```diff
@@ -0,0 +1,194 @@
+
+    Attributes:
+        function_names (List[str]): List of function names in 'catalog.schema.function' format.
+        tools_dict (Dict[str, FunctionTool]): A dictionary mapping function names to their
+            corresponding tools.
+        client (Optional[BaseFunctionClient]): The client used to manage functions.
+        description_updated (Optional[Bool]): Flag to check if the description has been updated.
+        cache_function (Optional[Callable]): Function that will be used to determine if the tool
+            should be cached, should return a boolean. If None, the tool will be cached.
+        result_as_answer (Optional[Bool]): Flag to check if the tool should be the final
+            agent answer.
+
+    """
+
+    function_names: List[str] = Field(
+        default_factory=list,
+        description="List of function names in 'catalog.schema.function' format",
+    )
+    tools_dict: Dict[str, CrewAIBaseTool] = Field(default_factory=dict)
+    client: Optional[BaseFunctionClient] = Field(
+        default=None,
+        description="Client for managing functions",
+    )
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    # CrewAI parameters, which can be found in the link below
+    # https://github.com/crewAIInc/crewAI-tools/blob/main/crewai_tools/tools/base_tool.py#L21
+    description_updated: Optional[bool] = Field(
+        default=False, description="Flag to check if the description has been updated."
+    )
+    cache_function: Optional[Callable] = Field(
+        default=lambda _args, _result: True,
+        description=(
+            "Function that will be used to determine if the tool should be cached, should return "
+            "a boolean. If None, the tool will be cached."
+        ),
+    )
+    result_as_answer: Optional[bool] = Field(
+        default=False, description="Flag to check if the tool should be the final agent answer."
+    )
+
+    @model_validator(mode="after")
+    def validate_toolkit(self) -> "UCFunctionToolkit":
+        self.client = validate_or_set_default_client(self.client)
+
+        if not self.function_names:
+            raise ValueError("Cannot create tool instances without function_names being provided.")
+
+        self.tools_dict = process_function_names(
+            function_names=self.function_names,
+            tools_dict=self.tools_dict,
+            client=self.client,
+            uc_function_to_tool_func=self.uc_function_to_crewai_tool,
+            description=self.description_updated,
+            cache_function=self.cache_function,
+            result_as_answer=self.result_as_answer,
+        )
+        return self
+
+    @staticmethod
+    def uc_function_to_crewai_tool(
+        *,
+        client: Optional[BaseFunctionClient] = None,
+        function_name: Optional[str] = None,
+        function_info: Optional[Any] = None,
+        description_updated: Optional[bool] = False,
```

Comment:
```suggestion
        description: Optional[bool] = False,
```
As on line 107 you passed description and it'll be passed here as kwargs

Reply:
Changed to be description_updated downstream. Great catch!